src/query/expression/src/types/binary.rs

```diff
@@ -456,6 +454,22 @@ impl BinaryColumnBuilder {
         self.data.get_unchecked(start..end)
     }
 
+    pub fn push_repeat(&mut self, item: &[u8], n: usize) {
+        if self.need_estimated && self.offsets.len() - 1 < 64 {
+            self.data.reserve(item.len() * n);
+            for _ in 0..n {
+                self.data.extend_from_slice(item);
+                self.commit_row();
+            }
+        } else {
+            let start = self.data.len();
+            let len = item.len();
+            self.data.extend(item.iter().cloned().cycle().take(len * n));
```

Comment:
Does it perform better than `data.extend_from_slice(scalar);`? If `n` is 1, there will be a clone for `[u8]`

Reply:
It's a mistake.